src/nile/core/deploy.py

```diff
@@ -39,3 +54,52 @@ def deploy(
 
     deployments.register(address, register_abi, network, alias)
     return address, register_abi
+
+
+def deploy_contract(
+    account,
+    contract_name,
+    salt,
+    unique,
+    calldata,
+    alias,
+    deployer_address,
+    max_fee,
+    overriding_path=None,
+    abi=None,
+):
+    """Deploy StarkNet smart contracts through UDC."""
+    logging.info(f"🚀 Deploying {contract_name}")
+
+    base_path = (
+        overriding_path if overriding_path else (BUILD_DIRECTORY, ABIS_DIRECTORY)
+    )
+    register_abi = abi if abi is not None else f"{base_path[1]}/{contract_name}.json"
+    deployer_for_address_generation = 0
+
+    if unique:
+        # Match UDC address generation
+        salt = compute_hash_chain(data=[account.address, salt])
+        deployer_for_address_generation = deployer_address
+
+    class_hash = get_class_hash(contract_name=contract_name)
+
+    address = calculate_contract_address_from_hash(
+        salt, class_hash, calldata, deployer_for_address_generation
+    )
```

Comment:
Indeed. `get_class_hash` should return an int to comply with the hex/int pattern we've just documented. @martriay I decided (open to discussion of course) to require returning values as int (not hex), because if not, we would be basically parsing in many parts of the code, because internal API requires to receive ints. Note this doesn't affect output files, or logging, just function return values.

Reply:
Fixed.